lib/clamav/client.rb

```diff
@@ -49,5 +49,18 @@ def resolve_default_socket
         ::UNIXSocket.new(unix_socket || '/var/run/clamav/clamd.ctl')
       end
     end
+
+    def ping
+      execute Commands::PingCommand.new
+    end
+
+    def scan(file)
+      execute Commands::InstreamCommand.new(file) if file.class = StringIO
+      execute Commands::ScanCommand.new(file)
+    end
+
+    def safe?(file)
```

Comment:
`scan` is likely to return an array right now, so calling `#virus_name` here is likely to raise an exception.